Sprint 17

usvladis opened the pull request:
Проделана работа по 17 спринту
Обновление от 28.09.2024:
-При редактировании названия категории клавиатура скрывается по нажатию на свободную область экрана
-Слово день/дней локализовано
-При переключении между днями недели действие фильтра сохраняется
-Реализованы скриншот тесты
-Если при переключении фильтра трекеры не найдены нужно появляется заглушка «Ничего не найдено»

usvladis committed:
Добавлена кнопка "Фильтры" на главный экран
По нажатию на кнопку осуществляется переход на экран выбора фильтров для главного экрана трекеров

usvladis committed:
Реализовали функционал удаления категорий

usvladis committed:
Добавил функционал редактирования названий категорий

usvladis committed:
Реализовал удаление трекеров
Убрал баг повторного создания трекера(Они дублировались в CoreData)
Исправил баг удаления категории. При удалении категории не удалялись трекеры из этой категории

usvladis committed:
Реализовал функционал редактирования трекеров, но! плохо работает редактирование категории у трекера. Вместо изменения категории мы меняем название категории...

usvladis committed:
Добавил возможность закреплять и откреплять трекеры
Что нужно поправить:
-Можно добавть анимации при удалении, закреплении и откреплении трекеров

usvladis committed:
Реализовал кнопку "Фильтры"

usvladis committed:
Сделал локализацию и темную тему для главного экрана
Позже нужно сделать это для всего проекта

usvladis committed:
-Добавил метрику YandexMobileMetrica
Добавил отслеживание действий пользователя для главного экрана

-Повсеместно добавил темную тему. Теперь приложение имеет темную тему полностью на всех доступныхэкранах

usvladis committed:
- Доработан экран статистики
Отображается статистика по завершенным трекерам

LizaIlinykh commented on CategoriesViewController/EditCategoryViewController.swift:
```diff
@@ -0,0 +1,112 @@
+    }
+    
+    private func setupCategoryView() {
+        //navigationItem.hidesBackButton = true
```
Рекомендую удалять лишний код

LizaIlinykh commented on CoreData/TrackerStore.swift:
```diff
@@ -30,10 +30,69 @@ final class TrackerStore {
+      guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return [] }
+      let managedContext = appDelegate.persistentContainer.viewContext
+      let fetchRequest = NSFetchRequest<TrackerCD>(entityName: "TrackerCD")
+      let trackerCoreDataArray = try! managedContext.fetch(fetchRequest)
```
Давай избавимся от принудительных операций, заменив их на optional binding

LizaIlinykh commented on TackerViewController/TrackerViewController.swift:
```diff
@@ -254,3 +322,121 @@ extension TrackerViewController: CreateTrackerDelegate {
+       private func updateUIWithVisibleTrackers(_ visibleTrackers: [TrackerCategory]) {
+           // Здесь обновляется пользовательский интерфейс с новыми отфильтрованными данными
+           collectionView.reloadData()
+       }
```
стоит сделать выравнивание во всех файлах. Выдели код cmd+A , затем примени выравнивание ctrl+I

LizaIlinykh commented on TackerViewController/TrackerViewModel.swift:
```diff
@@ -169,6 +192,159 @@ class TrackerViewModel {
+//            if let index = categories.firstIndex(where: { $0.title == "Закрепленные" }) {
+//                categories[index].trackers.append(tracker)
+//            }
+//        }
```
рекомендую удалять лишний код

usvladis committed:
Внес следующие измнения:
-При переключении между днями недели действие фильтра  сохраняются
-Если при переключении фильтра трекеры не найден показывается заглушка «Ничего не найдено»
-Если какой-то вариант фильтрации уже был выбран, при открытии списка фильтров он должен быть отмечен синей галочкой

usvladis committed:
Количество дней в треках на главном экране отображается как days или day — в зависимости от количества.

usvladis committed:
Доработал форматтер дней, добавил кейс для записи "0 дней"

usvladis committed:
Добавлены скриншот тесты
Первый тест проваливается, последующие тесты проходят усешно
